DNN Platform/Library/Services/Mail/MailKitMailProvider.cs

```diff
@@ -39,12 +39,28 @@ public override string SendMail(MailInfo mailInfo, SmtpInfo smtpInfo = null)
                 var mailMessage = CreateMailMessage(mailInfo, smtpInfo);
 
                 using (var smtpClient = new SmtpClient())
-                {
+                {
+                    var usingOAuth = smtpInfo.Authentication == "3";

```

Comment:
WOuld it be possible to combine this logic into a method to avoid having multiple locations with the same conditional connection logic?